src/features/cart/ShoppingCartTest.ts

```diff
@@ -4,171 +4,176 @@ interface Item{
     title:string,
     unitPrice:number,
     quantity:number,
-    discount:boolean
+    discount:string[]
 };
 
 
 interface DiscountCode{
     code:string,
+    type: "cart" | "item",
     value:number,
-    applied:string
+    itemId?: number;
 
 };
+interface Cart{
+    items:Item[],
+    discount:string[]
+}
 
-const discountCode:Array<DiscountCode>=[
+const discountCodes:Array<DiscountCode>=[
     {
         code:'NewYear2022',
+        type:"cart",
         value:20,
-        applied:'all'
     },
     {
         code:'summer2022',
+        type:"cart",
         value:10,
-        applied:'Mens-Plo T-shirt'
     },
     {
         code:'fullFree',
-        value:100,
-        applied:"Womens-Plo T-shirt"
+        type:"item",
+        value:70,
+        itemId:2
+        
     }
 
 ]
-const item:Item={
-    id:1,
-    title:'Mens TShirt',
-    unitPrice:35.99,
-    quantity:1,
-    discount:false
+const cart:Cart={
+    items:[],
+    discount:[]
 }
 
-let storeItem:Array<Item>=[];
-
 //Add to Product Card 
 const addToCart=(item:Item):void=>{
-    const product=storeItem.find((product)=>product.id==item.id);
-    if(product){
+    const isItem=cart.items.find((product)=>product.id===item.id)
+    if(isItem){
         item.quantity++
-        console.log(" item already exists",product);
-    }
-    else{
-        storeItem.push(item);
-        console.log("New Product add",storeItem.find((product)=>product.id==item.id));
-    }
-};
-addToCart({
-    id:1,
-    title:'Womens T-shirt',
-    unitPrice:39.99,
-    quantity:1,
-    discount:false
-});
-addToCart({
-    id:2,
-    title:'Mens T-shirt',
-    unitPrice:35.99,
-    quantity:1,
-    discount:false
-});
-addToCart({
-    id:3,
-    title:'Mens-Plo T-shirt',
-    unitPrice:20.99,
-    quantity:1,
-    discount:false
-});
-addToCart({
-    id:4,
-    title:'Womens-Plo T-shirt',
-    unitPrice:39.99,
-    quantity:1,
-    discount:false
-});
-//Add an Item form card 
-
-const increaseItem=(id:number):void=>{
-    const increaseQuantity=storeItem.find((item)=>item.id==id)
-    if(increaseQuantity){
-        increaseQuantity.quantity++;
-        console.log("Increse Quantity For this product ",increaseQuantity)
+    }else{
+        cart.items.push(item)
+        console.log("New Item Add" ,cart.items.find((product)=>product.id==item.id),)
     }
 };
+const removeToCart=(id:number):void=>{
+    const isItem=cart.items.find((product)=>product.id==id && product.quantity>=1)
+    if(isItem.quantity>1){
+        isItem.quantity--,
+        console.log("One Item Remove ",isItem)
 
-increaseItem(1);
-//An item remove from Card 
+    }
+    else if(isItem.quantity=1){
+        cart.items.filter((item)=>item.id !==isItem.id)
```

Comment:
only filtering cart.items array will not make change in main array. A solution for this may be to assign filtered array to cart.items.

for example,

> cart.items = cart.items.filter((item)=>item.id !==isItem.id)

Reply:
Thanks For review this Code I am solve this problem